src/main/java/com/synopsys/integration/detect/lifecycle/boot/autonomous/model/ScanType.java

```diff
@@ -0,0 +1,36 @@
+package com.synopsys.integration.detect.lifecycle.boot.autonomous.model;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class ScanType {
+    private String scanType;
+    private Map<String, String> scanProperties = new HashMap<>();
+    private Set<String> scanTargets = new HashSet<>();
```

Comment:
Could use SortedSet.

Reply:
If this is for improving readability when writing to JSON file, should `globalDetectProperties` and `detectorSharedProperties` also be sorted maps?

Reply:
Yes, good point!